env0/resource_user_project_assignment_test.go

```diff
@@ -91,6 +116,51 @@ func TestUnitUserProjectAssignmentResource(t *testing.T) {
 		})
 	})
 
+	t.Run("Create assignment and update custom role", func(t *testing.T) {
+		testCase := resource.TestCase{
+			Steps: []resource.TestStep{
+				{
+					Config: resourceConfigCreate(resourceType, resourceName, map[string]interface{}{
+						"user_id":        userId,
+						"project_id":     projectId,
+						"custom_role_id": customRole,
+					}),
+					Check: resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttr(accessor, "id", id),
+						resource.TestCheckResourceAttr(accessor, "user_id", userId),
+						resource.TestCheckResourceAttr(accessor, "project_id", projectId),
+						resource.TestCheckResourceAttr(accessor, "custom_role_id", customRole),
+						resource.TestCheckNoResourceAttr(accessor, "role"),
+					),
+				},
+				{
+					Config: resourceConfigCreate(resourceType, resourceName, map[string]interface{}{
+						"user_id":        userId,
+						"project_id":     projectId,
+						"custom_role_id": updatedCustomRole,
+					}),
+					Check: resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttr(accessor, "id", id),
+						resource.TestCheckResourceAttr(accessor, "user_id", userId),
+						resource.TestCheckResourceAttr(accessor, "project_id", projectId),
+						resource.TestCheckResourceAttr(accessor, "custom_role_id", updatedCustomRole),
+						resource.TestCheckNoResourceAttr(accessor, "role"),
+					),
+				},
+			},
+		}
+
+		runUnitTest(t, testCase, func(mock *client.MockApiClientInterface) {
+			gomock.InOrder(
+				mock.EXPECT().AssignUserToProject(projectId, &createCustomPayload).Times(1).Return(&createCustomResponse, nil),
+				mock.EXPECT().UserProjectAssignments(projectId).Times(2).Return([]client.UserProjectAssignment{otherResponse, createCustomResponse}, nil),
```

Comment:
Why should it be called twice?

Reply:
There is a "READ" after apply and there is a "READ" before update. (It's how terraform works).